packages/publisher/gcs/src/PublisherGCS.ts

```diff
@@ -0,0 +1,90 @@
+import path from 'path';
+
+import { PublisherBase, PublisherOptions } from '@electron-forge/publisher-base';
+import { Storage } from '@google-cloud/storage';
+import debug from 'debug';
+
+import { PublisherGCSConfig } from './Config';
+
+const d = debug('electron-forge:publish:gcs');
+
+type GCSArtifact = {
+  path: string;
+  keyPrefix: string;
+  platform: string;
+  arch: string;
+};
+
+export default class PublisherGCS extends PublisherBase<PublisherGCSConfig> {
+  name = 'gcs';
+
+  async publish({ makeResults, setStatusLine }: PublisherOptions): Promise<void> {
+    const { config } = this;
+    const artifacts: GCSArtifact[] = [];
+
+    if (!config.bucket) {
+      throw new Error('In order to publish to Google Cloud Storage you must set the "gcs.bucket" property in your Forge config.');
+    }
+
+    for (const makeResult of makeResults) {
+      artifacts.push(
+        ...makeResult.artifacts.map((artifact) => ({
+          path: artifact,
+          keyPrefix: config.folder || makeResult.packageJSON.version,
+          platform: makeResult.platform,
+          arch: makeResult.arch,
+        }))
+      );
+    }
+
+    const clientEmail = config.clientEmail || process.env.GOOGLE_CLOUD_CLIENT_EMAIL;
+    const privateKey = config.privateKey || process.env.GOOGLE_CLOUD_PRIVATE_KEY;
+
+    let credentials;
+    if (clientEmail && privateKey) {
+      credentials = {
+        client_email: clientEmail,
+        private_key: privateKey,
+      };
+    }
+
+    const storage = new Storage({
```

Comment:
issue: Since all auth options here are optional parameters in the Config, can we provide  some checks here to ensure that at least one set of auth options is present before calling `new Storage`?